### DISCUSSION TITLE
staline Themes

### DISCUSSION THREAD

**Discussion by Pakrohk** [nvim-lualine/lualine.nvim#970] [Configuration]
Hi, I am currently using StaLine.
But for some reasons (such as being more general and compatible with various themes and plugins), I would like to migrate to lualine.
Now the question I have is, how to apply the following config in lualine?

![image](https://user-images.githubusercontent.com/27810360/219226953-4fce6ee9-013a-4986-b4fe-123a2c21d446.png)

**Comment by fitrh** [comment]
What kind of config? The component or the background color? You can achieve the former by arranging the component order, for the latter, set a highlight group to have `guibg=NONE` and use it as your theme highlight

**Reply by Pakrohk** [reply]
> What kind of config? The component or the background color? You can achieve the former by arranging the component order, for the latter, set a highlight group to have `guibg=NONE` and use it as your theme highlight

For the first case (yes, it can be implemented easily), I'm mostly looking for a transparent background, but I don't want it to cause lualine to no longer follow the colorscheme.
You said to use the `guibg=NONE` flag, can you give an example for it?

**Reply by fitrh** [reply]
Something like this

```lua
local theme = {}

for _, mode in ipairs({ "normal", "insert", "visual", "replace", "command" }) do
  theme[mode] = {
    a = "StatusLine",
    b = "StatusLine",
    c = "StatusLine",
  }
end

theme.inactive = {
  a = "StatusLineNC",
  b = "StatusLineNC",
  c = "StatusLineNC",
}

require('lualine').setup({
  options = {
    theme = theme,
  },
})
```

All you need to do is adjust the highlight group to have `guibg=NONE`
